packages/macro/src/macroJs.ts

```diff
@@ -159,16 +173,29 @@ export default class MacroJs {
     // reset the expression counter
     this._expressionIndex = makeCounter()
 
-    const descriptor = this.processDescriptor(path.node.arguments[0])
+    let descriptor = this.processDescriptor(path.node.arguments[0])
+
+    if (process.env.NODE_ENV === "production") {
+      descriptor = this.stripNonEssentialProps(descriptor)
+    }
```

Comment:
Could you move this check into `processDescriptor` method and avoid code duplication in `replaceDefineMessage` and `replaceTAsFunction`?